shared/middleware/helpers/storage_apps.rb

```diff
@@ -14,7 +14,7 @@ def initialize(storage_id)
   def create(value, ip:, type: nil, published_at: nil)
     timestamp = DateTime.now
     row = {
-      storage_id: @storage_id,
+      storage_id: @storage_id.to_i,
```

Comment:
Good find. Looks like changing that line to generate a random decimal number instead of a hexadecimal one will fix the current problem. 

However, since it looks like this number is now getting written to the DB, I'm not sure what the consequences are of using a random number rather than using an ID of a valid row in the `storage_apps` table.

I'm not sure if this is enough information to unblock you so please ping me again if you'd like me to help dig a little deeper.